File: codes/DoublyLinkedList/Remove/Java/DoublyLinkedListRemove/DoublyLinkedListMain.java
package DoublyLinkedListRemove;

public class DoublyLinkedListMain {
    public static void main(String[] args) {
        DoublyLinkedList<Integer> list = new DoublyLinkedList<>();
        Integer[] data = new Integer[1];

        for (int i = 1; i < 9; i++)
            list.LInsert(i);

        if(list.LFirst(data)) {
            System.out.print(data[0]+" ");

            while(list.LNext(data))
                System.out.print(data[0]+" ");

            while(list.LPrevious(data))
                System.out.print(data[0]+" ");

            System.out.println("\n");
        }

        if (list.LFirst(data)) {
            if (data[0] % 2 == 0)
                list.LRemove();

            while (list.LNext(data))
                if (data[0] % 2 == 0)
                    list.LRemove();
        }

        if(list.LFirst(data)) {
            System.out.print(data[0]+" ");

            while(list.LNext(data))
                System.out.print(data[0]+" ");

            System.out.println("\n");
        }
    }
}

File: codes/DoublyLinkedList/NoneRemove/Python/DoublyLinkedListMain.py
import DoublyLinkedListModule as dlm

dlist = dlm.DoublyLinkedList()

for i in range(1, 9):
    dlist.LInsert(i)

data = dlist.LFirst()
if data[0] is True:
    print(data[1], end=' ')

    data = dlist.LNext()
    while data[0] is True:
        print(data[1], end=' ')
        data = dlist.LNext()
    
    data = dlist.LPrevious()
    while data[0] is True:
        print(data[1], end=' ')
        data = dlist.LPrevious()
    
    print()
    print()

data = dlist.LFirst()
if data[0] is True:
    if data[1] % 2 is 0:
        dlist.LRemove()
    
    data = dlist.LNext()
    while data[0] is True:
        if data[1] % 2 is 0:
            dlist.LRemove()
        data = dlist.LNext()

data = dlist.LFirst()
if data[0] is True:
    print(data[1], end=' ')

    data = dlist.LNext()
    while data[0] is True:
        print(data[1], end=' ')
        data = dlist.LNext()
    
    print()
    print()
